### ISSUE TITLE
Deleting a backing server bucket causes internal server error

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#1655]
Using Couchbase Server 4.1.0-5005 and Sync Gateway 1.2.1-3

Steps to reproduce
- Run sync_gateway with a backing server bucket. 

```
curl http://XXX.XX.XXX.XXX:4985
```

```
{"ADMIN":true,"couchdb":"Welcome","vendor":{"name":"Couchbase Sync Gateway","version":1.2},"version":"Couchbase Sync Gateway/1.2.1(3;0591221)"}
```

```
curl http://XXX.XX.XXX.XXX:4985/db/
```

```
{"committed_update_seq":602,"compact_running":false,"db_name":"db","disk_format_version":0,"instance_start_time":1458090656858885,"purge_seq":0,"state":"Online","update_seq":602}
```
- Delete server bucket

Expected behavior:
sync_gateway db should should be taken offline

Actual behavior

```
curl http://XXX.XX.XXX.XXX:4985/db/
```

```
{"error":"Internal Server Error","reason":"Internal error: EOF"}
```

**Comment by ajres** [comment]
@sethrosetter 

Is there an SG log available for this failure.

Has the test environment changed from 1.2.0, if yes could we re-run the same test for 1.2.0 on the updated environment.

**Comment by sethrosetter** [comment]
@ajres I will verify today. The tests have not changed, just the hardware they are running on.

**Comment by sethrosetter** [comment]
@ajres with sync_gateway 1.2.0-79

```
curl http://XXX.XX.XXX.XXX:4985/db/
```

```
{"committed_update_seq":602,"compact_running":false,"db_name":"db","disk_format_version":0,"instance_start_time":1458147191567184,"purge_seq":0,"state":"Online","update_seq":602}
```
- Delete server bucket

```
curl http://XXX.XX.XXX.XXX:4985/db/
```

```
{"committed_update_seq":0,"compact_running":false,"db_name":"db","disk_format_version":0,"instance_start_time":1458147191567184,"purge_seq":0,"state":"Offline","update_seq":0}
```

The db is taken offline as expected. I will post the logs shortly.

**Comment by sethrosetter** [comment]
```
2016-03-16T10:17:36.582-07:00 Changes: MultiChangesFeed done   (to seth)
2016-03-16T10:17:36.582-07:00 HTTP+: #814:     --> 200 OK  (0.0 ms)
2016-03-16T10:17:45.712-07:00 HTTP:  #815: GET /db/  (ADMIN)
2016-03-16T10:17:45.713-07:00 HTTP+: #815:     --> 200   (1.6 ms)
2016/03/16 10:18:10  Unable to decode response EOF
2016/03/16 10:18:10  Trying with http://172.23.105.125:8091/pools/default/bucketsStreaming/data-bucket
2016/03/16 10:18:10 Failed to connect to host, unexpected status code: 404. Body Requested resource not found.
2016/03/16 10:18:10  Trying with http://172.23.105.125:8091/pools/default/bucketsStreaming/data-bucket
2016/03/16 10:18:10 Failed to connect to host, unexpected status code: 404. Body Requested resource not found.
2016/03/16 10:18:10  Trying with http://172.23.122.250:8091/pools/default/bucketsStreaming/data-bucket
2016/03/16 10:18:10 Failed to connect to host, unexpected status code: 404. Body Requested resource not found.
2016/03/16 10:18:10  Trying with http://172.23.105.125:8091/pools/default/bucketsStreaming/data-bucket
2016/03/16 10:18:10 Failed to connect to host, unexpected status code: 404. Body Requested resource not found.
2016/03/16 10:18:10  Maximum failures reached. Exiting loop...
2016-03-16T10:18:10.485-07:00 WARNING: Bucket Updater for bucket data-bucket returned error: Max failures reached. Last Error Failed to connect to host. Status 404 Body Requested resource not found.
 -- base.GetCouchbaseBucket.func1() at bucket.go:469
2016-03-16T10:18:10.485-07:00 WARNING: Lost TAP feed for bucket data-bucket, with error: Max failures reached. Last Error Failed to connect to host. Status 404 Body Requested resource not found.
 -- rest.(*ServerContext)._getOrAddDatabaseFromConfig.func1() at server_context.go:646
2016-03-16T10:18:10.485-07:00 CRUD: Taking Database : db, offline
2016-03-16T10:18:10.485-07:00 CRUD: Waiting for all active calls to complete on Database : db
2016-03-16T10:18:10.485-07:00 CRUD: Database : db, is offline
2016/03/16 10:18:10  Bucket Updater exited with err Max failures reached. Last Error Failed to connect to host. Status 404 Body Requested resource not found.
2016/03/16 10:18:10 go-couchbase: TAP connection lost; reconnecting to bucket "data-bucket" in 1s
2016/03/16 10:18:11 go-couchbase: TAP connection lost; reconnecting to bucket "data-bucket" in 2s
2016/03/16 10:18:13 go-couchbase: TAP connection lost; reconnecting to bucket "data-bucket" in 4s
2016-03-16T10:18:14.874-07:00 HTTP:  #816: GET /db/  (ADMIN)
2016-03-16T10:18:14.875-07:00 WARNING: Error from Incr in sequence allocator (0) - attempt (1/3): EOF -- db.(*sequenceAllocator).incrWithRetry() at sequence_allocator.go:87
2016-03-16T10:18:14.887-07:00 WARNING: Error from Incr in sequence allocator (0) - attempt (2/3): EOF -- db.(*sequenceAllocator).incrWithRetry() at sequence_allocator.go:87
2016-03-16T10:18:14.898-07:00 WARNING: Error from Incr in sequence allocator (0) - attempt (3/3): EOF -- db.(*sequenceAllocator).incrWithRetry() at sequence_allocator.go:87
2016-03-16T10:18:14.909-07:00 WARNING: Too many failed Incr in sequence allocator - failing (0): EOF -- db.(*sequenceAllocator).incrWithRetry() at sequence_allocator.go:93
2016-03-16T10:18:14.909-07:00 WARNING: Error from Incr in lastSequence(): EOF -- db.(*sequenceAllocator).lastSequence() at sequence_allocator.go:39
2016-03-16T10:18:14.909-07:00 WARNING: Couldn't interpret error type *errors.errorString, value EOF -- base.ErrorAsHTTPStatus() at error.go:63
2016-03-16T10:18:14.909-07:00 HTTP: #816:     --> 500 Internal error: EOF  (35.4 ms)
2016-03-16T10:18:16.559-07:00 Cache+: Pruned 1 old entries from channel "*"
2016/03/16 10:18:17 go-couchbase: TAP connection lost; reconnecting to bucket "data-bucket" in 8s
2016-03-16T10:18:19.060-07:00 Cache+: Pruned 1 old entries from channel "*"
2016/03/16 10:18:25 go-couchbase: TAP connection lost; reconnecting to bucket "data-bucket" in 16s
2016-03-16T10:18:31.562-07:00 Cache+: Pruned 203 old entries from channel "*"
2016-03-16T10:18:31.563-07:00 Cache+: Pruned 203 old entries from channel "ABC"
2016-03-16T10:18:34.063-07:00 Cache+: Pruned 47 old entries from channel "*"
2016-03-16T10:18:34.063-07:00 Cache+: Pruned 47 old entries from channel "ABC"
2016/03/16 10:18:41 go-couchbase: TAP connection lost; reconnecting to bucket "data-bucket" in 30s

```

Full sg logs - https://gist.github.com/sethrosetter/72fa93e26d74adbe254e

**Comment by adamcfraser** [comment]
Fixed by #1657.
